app/lib/actions/katello/system/update.rb

```diff
@@ -23,8 +23,8 @@ def plan(system, sys_params)
           plan_action(::Actions::Katello::Foreman::HostUpdate, system)
           plan_action(::Actions::Pulp::Consumer::Update, system) if ::Katello.config.use_pulp
           plan_action(::Actions::Candlepin::Consumer::Update, system) if ::Katello.config.use_cp
-          plan_action(ElasticSearch::Reindex, system) if ::Katello.config.use_elasticsearch
           plan_action(::Actions::Candlepin::Consumer::RefreshSubscriptions, system, sys_params) if ::Katello.config.use_cp
+          plan_action(ElasticSearch::Reindex, system) if ::Katello.config.use_elasticsearch
```

Comment:
Moved elasticsearch after subscription refresh so candlepin data will be reflected properly